### ISSUE TITLE
Targeted update of multiversion package causes reinstall

### ISSUE THREAD

**Issue by jrohel** [openSUSE/libsolv#279]
Targeted update of multiversional package which is installed in topical version causes reinstall.

Example. Three multiversional packages are installed - `kernel-1-0.x86_64`, `kernel-2-0.x86_64`, and  `kernel-3-0.x86_64`. In available repo are packages `kernel-1-0.x86_64` and `kernel-3-0.x86_64`. Untargeted update does exactly what I want - nothink. **But targeted update does reinstall of topical package.**

```
repo system 0 testtags <inline>
#>=Pkg: kernel 1 0 x86_64
#>=Pkg: kernel 2 0 x86_64
#>=Pkg: kernel 3 0 x86_64
repo available 0 testtags <inline>
#>=Pkg: kernel 1 0 x86_64
#>=Pkg: kernel 3 0 x86_64
system x86_64 rpm system

job multiversion provides kernel
job update name kernel
result transaction,problems <inline>

nextjob

job multiversion provides kernel
job update name kernel [targeted]
result transaction,problems <inline>
#>reinstall kernel-3-0.x86_64@system kernel-3-0.x86_64@available
```

**Comment by mlschroe** [comment]
True. The current implementation of add_update_target() was a bit sub-optimal in the multiversion case. Fixed in commit 0ce04714c477f25fde16c57b5fdc478755794126

Please note that the current git master is changing in incompatible ways, so do not update to master for now. Instead, apply the commit as patch.

**Comment by jrohel** [comment]
@mlschroe Thanks
